messaging/src/main/java/org/axonframework/eventhandling/EventHandlerInvoker.java

```diff
@@ -16,12 +16,16 @@
 
 package org.axonframework.eventhandling;
 
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.ScopeAware;
+import org.axonframework.messaging.ScopeDescriptor;
+
 /**
  * Interface for an event message handler that defers handling to one or more other handlers.
  *
  * @author Rene de Waele
  */
-public interface EventHandlerInvoker {
+public interface EventHandlerInvoker extends ScopeAware {
```

Comment:
Good call to put the `ScopeAware` annotation on this level. That'll make it so that we can provide (deadline) `Message`s to any type of event handler.